core/src/main/java/com/sequenceiq/cloudbreak/reactor/DecommissionHandler.java

```diff
@@ -132,11 +132,25 @@ public void accept(Event<DecommissionRequest> event) {
                     .map(hostsToRemove::get)
                     .filter(Objects::nonNull)
                     .collect(Collectors.toList());
-            decommisionedInstances.forEach(clusterDecomissionService::deleteHostFromCluster);
-            updateInstanceStatuses(decommisionedInstances, InstanceStatus.DECOMMISSIONED, "instance successfully downscaled");
-            clusterDecomissionService.restartStaleServices();
+            String decommisionResult = decommisionedInstances.parallelStream().map(metaData -> {
```

Comment:
I think you should collect the result into `List<String>`. The `toString` method will format it well

Reply:
Will it cause issues with large clusters? Like when I downscale with 100 instances that will be 100 requests parallel?